.github/workflows/dh-backend-ci.yml

```diff
@@ -37,18 +39,35 @@ on:
   pull_request:
     branches:
       - main
-    paths:
-      - 'build/infrastructure/dh/**'
-      - 'apps/dh/api-dh/**'
-      - '.github/workflows/dh-backend-ci.yml'
-      - '.github/actions/publish-release-contents/action.yml'
 
 jobs:
-  dotnet_solution_ci:
-    uses: Energinet-DataHub/.github/.github/workflows/dotnet-solution-ci.yml@v8
+  #
+  # Detect changes to start relevant jobs
+  #
+
+  changes:
+    uses: ./.github/workflows/dh-backend-detect-changes.yml
+
+  #
+  # CI .NET
+  #
+
+  # Build all projects within solution
+  dotnet_ci_build:
+    needs: changes
+    if: ${{ needs.changes.outputs.dotnet == 'true' }}
+    uses: Energinet-DataHub/.github/.github/workflows/dotnet-build-prerelease.yml@v9
     with:
       SOLUTION_FILE_PATH: 'apps/dh/api-dh/DataHub.WebApi.sln'
-      PREPARE_OUTPUTS: true
+
+  # Run all tests in 'Tests.dll'
+  dotnet_ci_test_unittests:
+    needs: changes
+    if: ${{ needs.changes.outputs.dotnet == 'true' }}
+    uses: Energinet-DataHub/.github/.github/workflows/dotnet-solution-ci-test.yml@v8
+    with:
+      TESTS_DLL_FILE_PATH: '\apps\dh\api-dh\source\DataHub.WebApi.Tests\bin\Release\net6.0\Energinet.DataHub.WebApi.Tests.dll'
```

Comment:
If this comment is complete nonsense, then maybe I can make those changes in your branch. :)

We might have to use 'ASPNETCORE_TEST_CONTENTROOT_VARIABLE_NAME' and 'ASPNETCORE_TEST_CONTENTROOT_VARIABLE_VALUE' similar to what is shown here:
https://github.com/Energinet-DataHub/geh-charges/blob/main/.github/workflows/ci.yml#L106

Also add a comment to any class that inherit from 'WebApplicationFactory' similar to this: https://github.com/Energinet-DataHub/geh-charges/blob/e4702f151319f8b17e4f8570fee582e560b8d972/source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTest.Core/Fixtures/WebApi/WebApiFactory.cs#L27